torch/cuda/nccl.py

```diff
@@ -45,10 +48,27 @@ def all_reduce(inputs, outputs=None, op=SUM, streams=None, comms=None):
     torch._C._nccl_all_reduce(inputs, outputs, op, streams, comms)
 
 
-def reduce(inputs, outputs=None, root=0, op=SUM, streams=None, comms=None):
-    if outputs is None:
-        outputs = inputs
-    torch._C._nccl_reduce(inputs, outputs, root, op, streams, comms)
+# `output` used to be `outputs`, taking in a list of tensors. So we have two
+# arguments for BC reasons.
+def reduce(inputs, output=None, root=0, op=SUM, streams=None, comms=None, *, outputs=None):
+    if not isinstance(output, torch.Tensor) and isinstance(output, torch._six.container_abcs.Sequence):
+        # User called old API with positional arguments of list of output tensors.
+        if outputs is not None:
+            raise ValueError(
```

Comment:
Do we need to move this to a dedicated check? If an app specifies both output and outputs, it looks like output will be overwritten by outputs[root]?

Reply:
Ah good point. Will move!